src/main/java/com/contentgrid/gateway/security/jwt/issuer/jwk/source/FilebasedJWKSetSource.java

```diff
@@ -0,0 +1,101 @@
+package com.contentgrid.gateway.security.jwt.issuer.jwk.source;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.KeySourceException;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyUse;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.source.JWKSetCacheRefreshEvaluator;
+import com.nimbusds.jose.jwk.source.JWKSetParseException;
+import com.nimbusds.jose.jwk.source.JWKSetSource;
+import com.nimbusds.jose.jwk.source.JWKSetUnavailableException;
+import com.nimbusds.jose.proc.SecurityContext;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.ResourcePatternResolver;
+
+/**
+ * Loads JWKs from PEM-encoded private key files.
```

Comment:
```suggestion
 * Loads JWKs from PEM-encoded or JWK-encoded private key files.
```

It also supports loading JWKs from a JSON structure